node/node.go

```diff
@@ -187,14 +188,14 @@ func newTxValidator(pool mempool.Mempool, poolIDs *mempoolIDs, logger log.Logger
 			SenderP2PID: corep2p.ID(m.GetFrom()),
 			Context:     ctx,
 		})
-		switch err.(type) {
-		case mempool.ErrTxInCache:
+		switch {
+		case errors.Is(err, mempool.ErrTxInCache):
 			return true
-		case mempool.ErrMempoolIsFull:
+		case errors.Is(err, mempool.ErrMempoolIsFull{}):
 			return true
-		case mempool.ErrTxTooLarge:
+		case errors.Is(err, mempool.ErrTxTooLarge{}):
 			return false
-		case mempool.ErrPreCheck:
+		case errors.Is(err, mempool.ErrPreCheck{}):
```

Comment:
We should make the errors more consistent here IMO. But let's do that in a separate PR. All these do not need to be structs but can instead be annotated using `fmt`: https://pkg.go.dev/fmt#hdr-Format_errors